toucan_connectors/soap/soap_connector.py

```diff
@@ -73,11 +74,30 @@ def _retrieve_data(self, data_source: SoapDataSource) -> pd.DataFrame:
         # Force the casting of response
         response = getattr(client.service, data_source.method)(**data_source.parameters)
 
-        if isinstance(response, Sequence):
-            result = pd.DataFrame(response)
-        else:
+        if is_list_response(response):
```

Comment:
I would add before that `if/else` statement a comment about how and why the response can be a list of nested dict or a dict of lists

Reply:
:+1: I'll add that